docs/configuration/arguments.md

```diff
@@ -343,15 +343,16 @@ It also talks to a KVStore and has it's own copies of the same flags used by the
 
    Setting this to `true` enables writing to WAL during ingestion.
 
-- `-ingester.checkpoint-enabled`
-   Set this to `true` to enable checkpointing of in-memory chunks to disk. This is optional which helps in speeding up the replay process.
-
 - `-ingester.checkpoint-duration` 
    This is the interval at which checkpoints should be created.
 
 - `-ingester.recover-from-wal`
    Set this to `true` to recover data from an existing WAL. The data is recovered even if WAL is disabled and this is set to `true`. The WAL dir needs to be set for this.
 
+- `-ingester.checkpoint-enabled`
+   Set this to `true` to enable checkpointing of in-memory chunks to disk. This is optional which helps in speeding up the replay process.
+   (**Note:** Don't set it to `false` for anything other than small tests. There is no mechanism to delete the old WAL yet if checkpoint is disabled.).
```

Comment:
If setting it to `false` is discouraged, why is the default `false`? Shouldn't we change the default to `true`?

Reply:
Good point! Checkpoint should be enabled. I will make that change. But, does that count as a breaking change? :( (well, it's experimental, and not a big breaking change)

Reply:
Updated the default.